cli/src/commands/serve_web.rs

```diff
@@ -91,6 +91,12 @@ pub async fn serve_web(ctx: CommandContext, mut args: ServeWebArgs) -> Result<i3
 	if args.commit_id.is_none() {
 		cm.clone()
 			.start_update_checker(Duration::from_secs(update_check_interval));
+	} else {
+		// If a commit was provided, invoke get_latest_release() once to ensure we're using that exact version;
+		// get_latest_release() will short-circuit to args.commit_id.
+		if let Err(e) = cm.clone().get_latest_release().await {
```

Comment:
You should not need a clone here

```suggestion
		if let Err(e) = cm.get_latest_release().await {
```